### ISSUE TITLE
Repo maintained anymore?

### ISSUE THREAD

**Issue by BSmith-APFM** [microsoft/RulesEngine#604]
Just wanted to see if the authors of this repo are maintaining it anymore, or if its been left for the larger community.

**Comment by asulwer** [comment]
i have always felt this way.  there is little to no movement on this repo except to update dependencies

**Comment by asulwer** [comment]
i am actively using this project, i may create another repo and so it can be maintained properly

**Comment by abbasc52** [comment]
Hi All, I am no longer part of Microsoft and due to my current employment can no longer contribute to this project. I did contact forks at Microsoft but haven't heard back anything. If anyone plans to fork and maintain it, I can help with any knowledge required.

**Comment by BSmith-APFM** [comment]
> i am actively using this project, i may create another repo and so it can be maintained properly
If you fork it, I'll contribute.  I'd like to keep this project going, there are not many open source rules engine projects for dotnet out there!

**Comment by asulwer** [comment]
i created a fork and i have been actively making changes since.  mostly today

**Comment by asulwer** [comment]
> Just made the first release in my fork. Updated dependencies and updated to c# 12.0

Neat.  I have solved ALL issues in my fork.  I will be supporting new features and issues in mine.

**Comment by abbasc52** [comment]
@asulwer First off all great work! it seems you are determined to take it forward. Just a couple of advice:
- RulesEngine traditionally has been conservative towards making breaking changes, if we do, we give people some grace period to move to new implementation.
- Since your fork fixes couple of issues and adds new features(and potentially breaking changes), it would be great to add a migration guide(in case u release to nuget) . Users might eventually move to the fork but that would require a little hand holding.
- Also you might want to look into the .github folder. It has CI pipelines for PR check. and along its history it had full release pipeline (which was later removed and moved to azure pipelines). It can help setup the release process

**Comment by timophe-91** [comment]
@abbasc52  

Thank you for the template to the workflow. Just integrated  an updated version of it.

**Comment by RenanCarlosPereira** [comment]
Hey guys, we are using this nugget in one of our core applications… thats essencial we keep maintaining it, count on me on it.

Lets talk to see the future plans for it

Thanks for the great job so far 😄

**Comment by asulwer** [comment]
@abbasc52 we weren't sure if you were still active.  your linkedin profile states you are no longer with Microsoft.  we assumed this meant this wasn't going anywhere.  if you want to sit back and take a break thats fine, would you be willing to let a contributor move this forward? or maybe separate from Microsoft?  my planned direction will continue to make large breaking changes.  my fork wasn't to push you out, please do not think that.  issues are being resolved but not closed, just doing what i can.

**Comment by abbasc52** [comment]
@asulwer On the contrary, I want some to take it forward. I believe it is a great project and dont want it to die like this. I have been trying to get someone at Microsoft to take it forward but did not get much confirmation on it.

My comments earlier were just meant to ensure people looking for a maintained folk have a easy way out. If you have a different goal in mind, that is fine too :)

@RenanCarlosPereira , @timophe-91 , @RenanCarlosPereira  and @asulwer maybe you guys can come together and come up with a plan and a single fork? together it should be easier but you would need to agree on where this project should head 

I know it is asking too much out of you folks given I cant contribute anything anymore. Just a selfish wish to make sure this library remains useful to others :smile:

**Comment by timophe-91** [comment]
I would love to work together on this. But my issue is i don't want breaking changes so early and more in the long run with backwards  compatibility if possible. Since i'm using this in working environment and know it could be a pain in the ass to explain to any higher ups to migrate such stuff.

And since there are so few RulesEngine for .net it maybe could help too to split and it must not be a bad thing.

**Comment by timophe-91** [comment]
I'd love to collaborate on this project, but my main concern is avoiding frequent breaking changes, especially early on. Maintaining backward compatibility in the long run is crucial. Since I'm using this in a working environment, it would be difficult to justify (frequent) migrations to higher-ups. Given the limited number of Rules Engines available for .NET, it might be beneficial to explore splitting the project, as this doesn't necessarily have to be a negative outcome.

**Comment by RenanCarlosPereira** [comment]
@asulwer we can do it.
I don't want any breaking changes any soon as we are using it in some big projects.

I just have only one big concern thats come from this bug 🥺

https://github.com/zzzprojects/System.Linq.Dynamic.Core/issues/821

Lets keep this going with minimal breaking changes as possible.

If you guys want we can set up a quick meeting to talk about forks and other stuffs

**Comment by asulwer** [comment]
@RenanCarlosPereira if RulesEngine is affected please move this to an issue.

**Comment by RenanCarlosPereira** [comment]
Yeah it’s affecting it, I will open a bug, with an example of unit test on how its affecting it, thanks

**Comment by asulwer** [comment]
change is inevitable.  at the request of users, breaking changes occur in my fork but...current incorrectly created methods have been marked obsolete to support old code.  i am moving in a new direction with my fork, so is it a fork? for now yes but in the future no.

**Comment by RenanCarlosPereira** [comment]
> @RenanCarlosPereira if RulesEngine is affected please move this to an issue.

opened a new bug for this: 
https://github.com/microsoft/RulesEngine/issues/614

**Comment by jerbersoft** [comment]
@abbasc52 is there a continuation plan for this project? if you are not working for this project, how are we able to move forward but still making sure that we maintain the correct practice of PR's, code reviews and checks? And who might be the best candidates as the gate keepers of this codebase?

**Comment by asulwer** [comment]
@jerbersoft did you take a look at my fork?  all of the listed issues have been resolved there.  i wasn't opening PR's as i should but that has changed

**Comment by timophe-91** [comment]
@jerbersoft there are different forks now.

**Comment by RenanCarlosPereira** [comment]
We will try to maintain it in the @asulwer branch
We are going to keep minimal breaking changes.

Please feel free to join as a contributor
Only PRs are allowed, we have 2 breaking changes, not planning to introduce any other breaking changes (not for now)
If we do we will ask the comunity and discuss it together before any release or major changes

**Comment by RenanCarlosPereira** [comment]
@timophe-91, I have reviewed the code changes from 6.0.0 to 6.0.3 and its safe to make the move

Please be aware about those 2 breaking changes in the 6.0.0, other than that all the behavior remains the same.

**Comment by timophe-91** [comment]
@RenanCarlosPereira I've got my reasons to not trust that fork. But here is not the place to talk about those things. :) As said in another place, i stay with my own fork. :)

**Comment by RenanCarlosPereira** [comment]
No problem @timophe-91 

I got it, anyway, I will try to keep everything in order there, in case you change your mind.

Cheers 🍻

**Comment by rklec** [comment]
I just found out this is unmaintained and opened https://github.com/microsoft/RulesEngine/issues/625 to mark it as such.

However, what is the active/best fork now? I found the following:
* ⁠https://github.com/timophe-91/RulesEngine
⁠* https://github.com/asulwer/RulesEngine

Could not you work together to have _one_ active fork?

**Comment by RenanCarlosPereira** [comment]
Hi @rklec,

Thanks for bringing this up. I’m co-maintaining the [RulesEngine fork](https://github.com/asulwer/RulesEngine) along with @asulwer. This fork is quite active, and we’re committed to addressing issues and updating it regularly. Feel free to contact us anytime — we’re here to help and will do our best to resolve any concerns as quickly as possible.

As for @timophe-91, he’s maintaining his fork independently. 
While I’m not entirely sure of his specific reasons, I’m sure he has his motivations for doing so.

For my part, I’m fully invested in the @asulwer fork, which I’m also using in projects for several large companies globally. This is the version I recommend for ongoing development and support.

Best regards,

**Comment by timophe-91** [comment]
> I just found out this is unmaintained and opened #625 to mark it as such.
> 
> However, what is the active/best fork now? I found the following:
> 
> * ⁠https://github.com/timophe-91/RulesEngine
> * https://github.com/asulwer/RulesEngine
> 
> Could not you work together to have _one_ active fork?

I'm not working in the fork of asulwer. As mentioned before i have my reasons to to not do that.

**Comment by YogeshPraj** [comment]
Sorry everyone! Please let me know any pull request needs to be merged

**Comment by YogeshPraj** [comment]
yes
